### PR TITLE
add lasagne

### PR DESCRIPTION
**Pull request by nick-youngblut**
Checklist

- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [X] Source is from official source
- [X] Package does not vend other packages
- [X] Build number is 0
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your
      recipe (see
      [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)
      for more details)
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by nick-youngblut** [comment]
I used `conda skeleton pypi lasagne`, but the url in the generated meta.yaml appears to be wrong, since I always get the error: `RuntimeError: Could not download https://pypi.io/packages/source/l/lasagne/lasagne-0.1.tar.gz`. That url doesn't seem to exist

File: recipes/lasagne/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps -vv"
+
+requirements:
+  host:
+    - python
+    - setuptools
+    - numpy
```
Comment (scopatz) [inline recipes/lasagne/meta.yaml]: This line can be removed, since you are not building against numpy

**Comment by scopatz** [comment]
The PyPI name is `Lasagne`, not `lasagne`

**Comment by nick-youngblut** [comment]
> The PyPI name is `Lasagne`, not `lasagne`

Yep, that was the problem. It would definitely be helpful if `conda skeleton pypi` could distinguish whether the package name was capitalized or not.

**Comment by nick-youngblut** [comment]
@scopatz  Any idea about how do deal with this dependency conflict:

```
RuntimeError: failed to parse packages from exception: The following specifications were found to be incompatible with each other:
Package python conflicts for:
setuptools -> python[version='2.7.*|3.5.*|3.6.*|>=2.7,<2.8.0a0|>=3.5,<3.6.0a0|>=3.6,<3.7.0a0|>=3.7,<3.8.0a0']
numpy -> mkl_fft[version='>=1.0.6,<2.0a0'] -> python=2.7
python[version='>=3.7,<3.8.0a0']
theano==0.8.2 -> numpy[version='>=1.3.0'] -> python[version='>=2.7,<2.8.0a0|>=3.5,<3.6.0a0|>=3.6,<3.7.0a0']
theano==0.8.2 -> python[version='2.7.*|3.5.*|3.6.*']
numpy -> python[version='3.5.*|3.6.*|>=2.7,<2.8.0a0|>=3.5,<3.6.0a0|>=3.6,<3.7.0a0|>=3.7,<3.8.0a0']
```

**Comment by jjhelmus** [comment]
> how do deal with this dependency conflict:

theano 0.8.2 is not available for Python 3.7.  Do you need that particular version or will newer versions of theano work?

**Comment by nick-youngblut** [comment]
@jjhelmus Thanks for clarifying the dependency conflict. Yes, lasagne does [require theano 0.8.2](https://github.com/Lasagne/Lasagne/blob/master/requirements.txt). Setting `python<=3.6` will hopefully do the trick.

**Comment by nick-youngblut** [comment]
@scopatz @jjhelmus Thanks again for your help with this recipe! Any ideas on what to do about this current `[WinError 123]` error?

**Comment by scopatz** [comment]
Looks like you are missing some deps

**Comment by nick-youngblut** [comment]
> Looks like you are missing some deps

I don't understand why missing deps would cause the error:

`OSError: [WinError 123] The filename, directory name, or volume label syntax is incorrect: 'C:\\bld\\lasagne_1568825063486\\work_moved_lasagne-0.1-py<=36_0_win-64_main_build_loop'`

**Comment by nick-youngblut** [comment]
@scopatz can you please merge my pull request?

**Pull request merged**